solr/core/src/test/org/apache/solr/servlet/HttpSolrCallCloudTest.java

```diff
@@ -93,40 +114,66 @@ private void assertCoreChosen(int numCores, TestRequest testRequest) throws Unav
     assertEquals(numCores, coreNames.size());
   }
 
-  private static class TestResponse extends Response {
+  public static class TestRequest implements HttpServletRequest {
```

Comment:
I hate mocks but as I look at this, mocking seems perfect so you don't have to implement everything.  Or maybe there's a base class as there was before that you can extend.  (Same feedback for another source file you did this in)